### PR TITLE
Fixes for Persian calendar implementation

### PR DESCRIPTION
**Pull request by jason-p-pickering**

### CODE REVIEW

**Comment by larshelge** [comment]
This PR suffers from hardcoding (PeriodType.getCalendar().name().equals( "persian" )), codestyle and conflicts with master. Closing for now.

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/calendar/impl/PersianCalendar.java [hunk 1]
```diff
@@ -196,9 +189,40 @@ public int weeksInYear( int year )
     @Override
     public int isoWeek( DateTimeUnit dateTimeUnit )
     {
-        DateTime dateTime = toIso( dateTimeUnit )
-            .toJodaDateTime( ISOChronology.getInstance( DateTimeZone.getDefault() ) );
-        return dateTime.getWeekOfWeekyear();
+
+        if (dateTimeUnit.getYear() > START_ISO.getYear()) {
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/calendar/impl/PersianCalendar.java]: Please correct the code style to apply to the DHIS 2 style with brackets on new line, spaces after ( ).

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/i18n/I18nFormat.java [hunk 2]
```diff
@@ -220,7 +221,18 @@ public String formatPeriod( Period period )
             return null;
         }
 
-        String typeName = period.getPeriodType().getName();
+        String typeName = period.getPeriodType().getName();        
+
+        if ( PeriodType.getCalendar().name().equals( "persian" ) )
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/i18n/I18nFormat.java]: Please remove the hard-coding of this specific calendar in I18nFormat. We would like I18nFormat to be unaware of any period types. This should be built into the Calendar solution.

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/PeriodType.java [hunk 3]
```diff
@@ -474,7 +475,8 @@ public Period createPeriod( DateInterval dateInterval )
         final DateTimeUnit from = cal.toIso( dateInterval.getFrom() );
         final DateTimeUnit to = cal.toIso( dateInterval.getTo() );
 
-        return new Period( this, from.toJdkDate(), to.toJdkDate(), getIsoDate( from ) );
+        return new Period( this, from.toJdkDate(), to.toJdkDate(), cal.name() == "persian" ? getIsoDate( dateInterval.getFrom() ) : getIsoDate( from ));
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/PeriodType.java]: Please remove the hard-coding of this specific calendar in PeriodType. We would like PeriodType to be unaware of any period types. This should be built into the Calendar solution.

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultDataQueryService.java [hunk 4]
```diff
@@ -97,11 +97,11 @@ public DataQueryParams getFromUrl( Set<String> dimensionParams, Set<String> filt
         boolean duplicatesOnly, String approvalLevel, Date relativePeriodDate, String userOrgUnit, boolean allowAllPeriods, DhisApiVersion apiVersion )
     {
         I18nFormat format = i18nManager.getI18nFormat();
-        
+
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultDataQueryService.java]: This branch now conflicts with master, please resolve those conflicts.

**Comment by jason-p-pickering** [comment]
Can one of the admins verify this patch?

**Comment by larshelge** [comment]
Closing due to inactivity.

**Pull request closed without merging**